src/passes/LegalizeJSInterface.cpp

```diff
@@ -185,6 +208,32 @@ struct LegalizeJSInterface : public Pass {
     return im->module == ENV && im->base.startsWith("invoke_");
   }
 
+  Function* tempSetter(Module* module) {
+    if (!setTempRet0) {
+      if (exportedHelpers) {
+        auto* ex = module->getExport(SET_TEMP_RET_EXPORT);
+        setTempRet0 = module->getFunction(ex->value);
+      } else {
+        setTempRet0 = getFunctionOrImport(
+          module, SET_TEMP_RET_IMPORT, Type::i32, Type::none);
+      }
+    }
+    return setTempRet0;
+  }
+
+  Function* tempGetter(Module* module) {
+    if (!getTempRet0) {
+      if (exportedHelpers) {
+        auto* ex = module->getExport(GET_TEMP_RET_EXPORT);
+        getTempRet0 = module->getFunction(ex->value);
+      } else {
+        getTempRet0 = getFunctionOrImport(
+          module, GET_TEMP_RET_IMPORT, Type::none, Type::i32);
+      }
+    }
+    return getTempRet0;
+  }
```

Comment:
Can use a function called by both of these to reduce the code duplication perhaps?

Reply:
The function have different signatures .. so its not trivial to use the same getting here.

Reply:
Yeah, I guess it's best as it is.